mypy/find_sources.py

```diff
@@ -92,7 +92,7 @@ def expand_dir(self, arg: str, mod_prefix: str = '') -> List[BuildSource]:
                 base, suffix = os.path.splitext(name)
                 if base == '__init__':
                     continue
-                if base not in seen and '.' not in base and suffix in PY_EXTENSIONS:
+                if base not in seen and suffix in PY_EXTENSIONS:
                     seen.add(base)
                     src = BuildSource(path, mod_prefix + base, None, base_dir)
```

Comment:
Passing in `mod_prefix + base` to `BuildSource` here is probably incorrect when there's a period in the `base` name. It might make sense to pass in `None`. Someone who knows build better would know for sure :-)

Reply:
Although when you explicitly list the file we pass that in as module name to `BuildSource` even if it contains a period, so maybe it's fine: https://github.com/python/mypy/blob/master/mypy/find_sources.py#L34

Reply:
@hauntsaninja Thanks for pointing this out, I think you are right, as I think it is also mentioned in pep guideline that using period is not allowed in file names, but then i tried it on the tests and it was working, so I am not sure how the build is configured. So yes someone with better experience with build would be knowing.